src/processes/follow-topic-process.js

```diff
@@ -31,31 +31,30 @@ const anon_to_sign_user = async (client, user, targetUser) => {
   });
   if (channel) {
     return channel;
-  } else {
-    const channel_id = generate_channel_id_for_anon_chat(user, targetUser, 'autoMessage');
-    const emoji = BetterSocialConstantListUtils.getRandomEmoji();
-    const color = BetterSocialConstantListUtils.getRandomColor();
-    const anon_init_data = {
-      anon_user_info_color_code: color.code,
-      anon_user_info_color_name: color.color,
-      anon_user_info_emoji_code: emoji.emoji,
-      anon_user_info_emoji_name: emoji.name
-    };
-
-    const new_channel = await ChatAnonUserInfo.create({
-      channel_id: channel_id,
-      my_anon_user_id: user,
-      target_user_id: targetUser,
-      anon_user_info_color_code: anon_init_data.anon_user_info_color_code,
-      anon_user_info_color_name: anon_init_data.anon_user_info_color_name,
-      anon_user_info_emoji_code: anon_init_data.anon_user_info_emoji_code,
-      anon_user_info_emoji_name: anon_init_data.anon_user_info_emoji_name,
-      context: 'auto message',
-      initiator: user,
-      source_id: ''
-    });
-    return new_channel;
   }
+  const channel_id = generate_channel_id_for_anon_chat(user, targetUser, 'autoMessage');
+  const emoji = BetterSocialConstantListUtils.getRandomEmoji();
+  const color = BetterSocialConstantListUtils.getRandomColor();
+  const anon_init_data = {
+    anon_user_info_color_code: color.code,
+    anon_user_info_color_name: color.color,
+    anon_user_info_emoji_code: emoji.emoji,
+    anon_user_info_emoji_name: emoji.name
+  };
+
+  const new_channel = await ChatAnonUserInfo.create({
+    channel_id,
+    my_anon_user_id: user,
+    target_user_id: targetUser,
+    anon_user_info_color_code: anon_init_data.anon_user_info_color_code,
+    anon_user_info_color_name: anon_init_data.anon_user_info_color_name,
+    anon_user_info_emoji_code: anon_init_data.anon_user_info_emoji_code,
+    anon_user_info_emoji_name: anon_init_data.anon_user_info_emoji_name,
+    context: 'auto message',
+    initiator: user,
+    source_id: ''
+  });
+  return new_channel;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The removal of the `else` clause and the indentation adjustment improves readability. The function will return early if the channel exists, so there's no need for an `else` clause. This is a good application of the "return early" principle. However, it's important to ensure that the `channel_id` variable is not used elsewhere in the function after this block of code, as its scope has been reduced to this block only.
```diff
- } else {
-   const channel_id = generate_channel_id_for_anon_chat(user, targetUser, 'autoMessage');
-   ...
-   return new_channel;
- }
+ const channel_id = generate_channel_id_for_anon_chat(user, targetUser, 'autoMessage');
+ ...
+ return new_channel;
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->